cirq-google/cirq_google/engine/abstract_local_program.py

```diff
@@ -0,0 +1,206 @@
+# Copyright 2021 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import copy
+import datetime
+from typing import Dict, List, Optional, Sequence, Set, TYPE_CHECKING, Union
+import cirq
+from cirq_google.engine.client import quantum
+from cirq_google.engine.abstract_program import AbstractProgram
+
+if TYPE_CHECKING:
+    from cirq_google.engine.abstract_local_job import AbstractLocalJob
+    from cirq_google.engine.abstract_local_engine import AbstractLocalEngine
+
+
+class AbstractLocalProgram(AbstractProgram):
+    """A quantum program designed for local in-memory computation.
+
+    This implements all the methods in `AbstractProgram` using
+    in-memory structions.  Labels, descriptions, and time are all
+    stored using dictionaries.
+
+    This is a partially implemented instance.  Inheritors will still
+    A program created via the Engine interface.
```

Comment:
There's a disconnect here - line 33 seems incomplete, and 34 seems redundant.

Reply:
Fixed.